craft_store/publisher/_request.py

```diff
@@ -154,22 +167,22 @@ class RevokeMacaroonRequest(BaseModel):
 class PackageLinks(BaseModel):
     """Links structure for package metadata."""
 
-    contact: list[str] | None = Field(
+    contact: Sequence[Annotated[str, annotated_types.MaxLen(2000)]] | None = Field(
```

Comment:
I've not actually used these abstract base classes with pydantic before. Could you add some unit tests to show that they validate and serialise correctly?